cpp/oneapi/dal/algo/kmeans/backend/cpu/infer_kernel.cpp

```diff
@@ -37,51 +37,69 @@ template <typename Float, daal::CpuType Cpu>
 using daal_kmeans_lloyd_dense_kernel_t =
     daal_kmeans::internal::KMeansBatchKernel<daal_kmeans::lloydDense, Float, Cpu>;
 
+inline auto get_daal_parameter_to_infer(const descriptor_t& desc) {
+    const std::int64_t max_iteration_count = 0;
+
+    daal_kmeans::Parameter parameter(
+        dal::detail::integral_cast<std::size_t>(desc.get_cluster_count()),
+        dal::detail::integral_cast<std::size_t>(max_iteration_count));
+
+    if (desc.get_result_options().test(result_options::compute_exact_objective_function)) {
+        parameter.resultsToEvaluate =
+            static_cast<DAAL_UINT64>(daal_kmeans::computeAssignments) |
+            static_cast<DAAL_UINT64>(daal_kmeans::computeExactObjectiveFunction);
+    }
+    else {
+        parameter.resultsToEvaluate = static_cast<DAAL_UINT64>(daal_kmeans::computeAssignments);
+    }
+
+    return parameter;
+}
+
 template <typename Float, typename Task>
 static infer_result<Task> call_daal_kernel(const context_cpu& ctx,
                                            const descriptor_t& desc,
                                            const model<Task>& trained_model,
                                            const table& data) {
     const std::int64_t row_count = data.get_row_count();
-    const std::int64_t cluster_count = desc.get_cluster_count();
-    const std::int64_t max_iteration_count = 0;
 
-    daal_kmeans::Parameter par(dal::detail::integral_cast<std::size_t>(cluster_count),
-                               dal::detail::integral_cast<std::size_t>(max_iteration_count));
-    par.resultsToEvaluate = static_cast<DAAL_UINT64>(daal_kmeans::computeAssignments) |
-                            static_cast<DAAL_UINT64>(daal_kmeans::computeExactObjectiveFunction);
+    auto result = infer_result<task::clustering>{}.set_result_options(desc.get_result_options());
```

Comment:
```suggestion
    auto result = infer_result<Task>{}.set_result_options(desc.get_result_options());
```
I guess task is always clustering? but it at least matches previous code

Reply:
good point will fix